grpc-protocol/src/main/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -127,31 +127,33 @@ private static void checkGrpcStatus(@Nullable String grpcStatus, HttpHeaders hea
         }
     }
 
-    private static final class GrpcFramingDecorator extends SimplePooledDecoratingHttpClient {
+    private static final class GrpcFramingDecorator extends SimpleDecoratingHttpClient {
 
         private GrpcFramingDecorator(HttpClient delegate) {
             super(delegate);
         }
 
         @Override
-        public HttpResponse execute(
-                PooledHttpClient client, ClientRequestContext ctx, PooledHttpRequest req) {
+        public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
             return HttpResponse.from(
                     req.aggregateWithPooledObjects(ctx.eventLoop(), ctx.alloc())
                        .thenCompose(
                                msg -> {
-                                   final ByteBuf buf = msg.content().content();
-                                   final PooledHttpData framed;
+                                   final ByteBuf buf;
+                                   if (msg.content() instanceof ByteBufHolder) {
```

Comment:
This is the pattern which is annoying currently unless we guarantee msg is always a `ByteBufHolder` or `PooledHttpData`, possibly by having non-pooled also implement the interface using no-op / wrapping

Reply:
Let me think about this and send a follow up pull request if I find something. Thanks for suggestion.